src/lib/subdevs/xocl-axigate.c

```diff
@@ -123,8 +123,12 @@ static void xocl_axigate_freeze(struct platform_device *pdev)
 	mutex_lock(&gate->gate_lock);
 	freeze = reg_rd(gate, iag_rd);
 	if (freeze) {		/* gate is opened */
+		/* use the same async broadcast calls to make sure the
+		 * order with gate free
+		 */
```

Comment:
sounds like you're depending on the fact that all async event is added to one single list? this is implementation detail and i don't feel comfortable for this kind of dependency. what is the race we're facing here? maybe we can find a more reliable way to solve it.